src/test/run-make-fulldeps/coverage-reports-base/expected_show_coverage.conditions.txt

```diff
@@ -36,34 +37,70 @@
    36|      0|        return;
    37|       |    }
    38|       |
-   39|      1|    let mut countdown = 0;
-   40|      1|    if true {
-   41|      1|        countdown = 1;
-   42|      1|    }
-   43|       |
-   44|      1|    let z = if countdown > 7 {
-                      ^0
-   45|      0|        countdown -= 4;
-   46|      1|    } else if countdown > 2 {
-   47|      0|        if countdown < 1 || countdown > 5 || countdown != 9 {
-   48|      0|            countdown = 0;
-   49|      0|        }
-   50|      0|        countdown -= 5;
-   51|       |    } else {
-   52|      1|        let should_be_reachable = countdown;
-   53|      1|        println!("reached");
-   54|      1|        return;
-   55|       |    };
+   39|      1|    if true {
+   40|       |        // Demonstrate the difference with `TerminatorKind::Assert` as of 2020-11-15. Assert is no
+   41|       |        // longer treated as a `BasicCoverageBlock` terminator, which changed the coverage region,
+   42|       |        // for the executed `then` block above, to include the closing brace on line 30. That
+   43|       |        // changed the line count, but the coverage code region (for the `else if` condition) is
+   44|       |        // still valid.
+   45|       |        //
+   46|       |        // Note that `if` (then) and `else` blocks include the closing brace in their coverage
+   47|       |        // code regions when the last line in the block ends in a semicolon, because the Rust
+   48|       |        // compiler inserts a `StatementKind::Assign` to assign `const ()` to a `Place`, for the
+   49|       |        // empty value for the executed block. When the last line does not end in a semicolon
+   50|       |        // (that is, when the block actually results in a value), the additional `Assign` is not
+   51|       |        // generated, and the brace is not included.
+   52|      1|        let mut countdown = 0;
```

Comment:
Yeah, this is pretty unintuitive. "Closing brace is counted when the scope is exited, or skipped by if/loop condition," seems like the most logical behavior to me.

Reply:
> "Closing brace is counted when the scope is exited, or skipped by if/loop condition,"

@tmandry are you quoting something here, or just suggesting a way you'd like it described?

On the one hand, counting the closing brace when exiting the scope is how we currently count the closing brace of a function, so this would be more consistent with that.

An argument could be made that I shouldn't count the closing brace of a function, though, if the function was exited by an early `return`.

The truth is, in some cases, the reason things work the way they do for functions and for loops is because it was the best I could do given the MIR, and seemed "reasonable" (even if there were alternatives like these).

Personally, I like the idea of changing how function end braces are counted, rather than changing how loop end braces are counted. I think it's more intuitive to count the end brace only if the code did not exit the scope earlier (by break, continue, or return).

Either way, it's inconsistent now.

Should we continue debating this here? Or log it as an unresolved new "Issue" and invite the community for comment?

If the latter, maybe I shouldn't change either one, for now, since it could be a waste of time if we decide to go a different way.

Reply:
Yeah I think there are a few competing principles..

- Consistency
- Intuitiveness
- Preserving all information

and I'm not exactly sure how to resolve them. My quote was one idea I had.

This might be good to hash out in an issue somewhere.

Reply:
I think a lot of things like this are resolved with my most recent improvements.